Sources/StevensonCore/GitHubService.swift

```diff
@@ -0,0 +1,63 @@
+import Foundation
+import Vapor
+
+public struct GitHubService {
+    private let baseUrl = "https://api.github.com:443"
+    private let headers: HTTPHeaders
+
+    public init(username: String, token: String) {
+        let base64Auth = Data("\(username):\(token)".utf8).base64EncodedString(options: [])
+        self.headers = [
+            "Authorization": "Basic \(base64Auth)",
+            "Accept": "application/json"
+        ]
+    }
+}
+
+public extension GitHubService {
+    public typealias CommitRef = String
```

Comment:
Would `CommitSHA` be a better name? We could use Tagged for this as well.

Reply:
That's not necessary the SHA though, it can be a branch name too. Maybe CommitTree would be better?

I'm not sure about using Tagged here, as it's not really gonna be used for "identifiers" there.
Maybe I should just remove the typealias and use String (used it before my multiple refactoring when my draft API was not that clean but now it's might not be as useful)

Reply:
Typealias is good if it makes it more clear what this can be. Otherwise just a comment on the property would be enough, i.e. here it can say "git reference (i.e. she, branch reference or any other valid git reference)"